lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs

```diff
@@ -1216,6 +1222,36 @@ toStakePoolDlgCert xpub (W.PoolId pid) =
     cred = SL.KeyHash $ UnsafeHash $ toShort $ blake2b224 $ xpubPublicKey xpub
     pool = SL.KeyHash $ UnsafeHash $ toShort pid
 
+{-------------------------------------------------------------------------------
+                   Assessing sizes of token bundles
+-------------------------------------------------------------------------------}
+
+-- | Assesses a token bundle size in relation to the maximum size that can be
+--   included in a transaction output.
+--
+assessTokenBundleSize :: TokenBundle.TokenBundle -> W.TokenBundleSizeAssessment
+assessTokenBundleSize tb
+    | serializedLengthBytes <= maxTokenBundleSerializedLengthBytes =
+        W.TokenBundleSizeWithinLimit
+    | otherwise =
+        W.TokenBundleSizeExceedsLimit
+  where
+    serializedLengthBytes :: Int
+    serializedLengthBytes = computeTokenBundleSerializedLengthBytes tb
+
+computeTokenBundleSerializedLengthBytes :: TokenBundle.TokenBundle -> Int
+computeTokenBundleSerializedLengthBytes =
+    BS.length . Binary.serialize' . Cardano.toMaryValue . toCardanoValue
+
+-- NOTE: This hard-coded limit may change in future. Ideally, we should
+-- delegate the assessment of whether a token bundle is too large to a
+-- function exported by Cardano API.
+--
+-- See: https://jira.iohk.io/projects/ADP/issues/ADP-779
+--
+maxTokenBundleSerializedLengthBytes :: Int
+maxTokenBundleSerializedLengthBytes = 4096
```

Comment:
This should be `4000` (4 kB), and not `4096` (4 KiB).

Source: https://github.com/input-output-hk/cardano-ledger-specs/blob/5bccddb0ef0a1ae270b72fe0782a8ad680ed5852/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/Rules/Utxo.hs#L310

Reply:
Fixed in 82acfd0ca91e2859bc61d51cb51cff121b36fdee.